src/travel-details-screens/utils.ts

```diff
@@ -106,9 +111,146 @@ export function significantWaitTime(seconds: number) {
   return seconds > MIN_SIGNIFICANT_WAIT_IN_SECONDS;
 }
 
-export function getLineName(leg: Leg) {
+export type FavoriteDepartureMigrationPair = {
+  lineName?: string;
+  destinationDisplay?: DestinationDisplay;
+};
+
+const getUniqueFavoriteDepartureMigrationPairs = (
+  stopPlaceGroups: StopPlaceGroup[],
+): FavoriteDepartureMigrationPair[] => {
+  const nestedFavoriteDepartureMigrationPairs = stopPlaceGroups.map(
+    (stopPlaceGroup) =>
+      stopPlaceGroup?.quays.map((quay) =>
+        quay.group.map((groupItem) => ({
+          lineName: groupItem?.lineInfo?.lineName,
+          destinationDisplay: groupItem?.lineInfo?.destinationDisplay,
+        })),
+      ),
+  );
+  // flatten 3d array to 1d and ensure unique migration pairs
+  return uniqBy(
+    flatten(flatten(nestedFavoriteDepartureMigrationPairs)),
+    (pair) => pair.lineName,
+  );
+};
+
+const shouldStoredFavoriteDepartureBeMigrated = (
+  storedFavDep: StoredFavoriteDeparture,
+  favDepMigrationPair: FavoriteDepartureMigrationPair,
+): boolean => {
+  const storedFavDepDD = storedFavDep?.destinationDisplay;
+  const storedFrontText = storedFavDepDD?.frontText;
+
+  const {lineName, destinationDisplay} = favDepMigrationPair;
+
+  const storedFrontTextEqualsLineName = storedFrontText === lineName;
+  const storedFrontTextIncludesVia = !!storedFrontText?.includes(' via ');
+  const storedFavDepDDViaIsEmpty = (storedFavDepDD?.via?.length || 0) === 0;
+  const destinationDisplayViaIsNotEmpty =
+    (destinationDisplay?.via?.length || 0) > 0;
+  return (
+    storedFrontTextEqualsLineName &&
+    storedFrontTextIncludesVia &&
+    storedFavDepDDViaIsEmpty &&
+    destinationDisplayViaIsNotEmpty
+  );
```

Comment:
What happens if the user has two favorites, "Lade via Strindheim" and "Lade via Tunga"? 🤔 Let's talk next week.

Reply:
`storedFrontTextEqualsLineName` will ensure there is no cross-matching in that case. `lineName` comes from the bff and contains the whole string